src/dashboard/dashboard.js

```diff
@@ -1,10 +1,77 @@
 import React from "react";
+import ChatListComponent from "../chatlist/chatList";
+import { Button, withStyles } from "@material-ui/core";
+import styles from "./styles";
+import ChatViewComponent from "../chatview/chatView";
+import firebase from "../firebase";
 
 class DashboardComponent extends React.Component {
-    
-    render() {
-        return(<div>Hello World from Dashboard</div>)
-    }
+  constructor() {
+    super();
+    this.state = {
+      selectedChat: null,
+      newChatFormVisible: false,
+      email: null,
+      chats: [],
+    };
+  }
+
+  selectChat = (chatIndex) => {
+    this.setState({ selectedChat: chatIndex });
+  };
+
+  newChatBtnClicked = () => {
+    this.setState({ newChatFormVisible: true, selectedChat: null });
+  };
+
+  componentDidMount = () => {
+    firebase.auth().onAuthStateChanged(async (_usr) => {
+      if (!_usr) {
+        this.props.history.push("/login");
+      } else {
+        await firebase
+          .firestore()
+          .collection("chats")
+          .where("users", "array-contains", _usr.email)
+          .onSnapshot(async (res) => {
+            const chats = res.docs.map((_doc) => _doc.data());
+            await this.setState({
+              email: _usr.email,
+              chats: chats,
+            });
+            console.log(this.state);
+          });
+      }
+    });
+  };
+
+  signOut = () => firebase.auth().signOut();
+
+  render() {
+    const { classes } = this.props;
+
+    return (
+      <div>
+        <ChatListComponent
+          history={this.props.history}
+          newChatBtnFn={this.newChatBtnClicked}
+          selectChatFn={this.selectChat}
+          chats={this.state.chats}
+          userEmail={this.state.email}
+          selectedChatIndex={this.state.selectedChat}
+        ></ChatListComponent>
```

Comment:
If a React component has no children (like this one) you don't need to add the matching closing tag. You could write it like this instead;

```
<ChatListComponent
...all of your props
/>
```
Notice that it doesn't have the closing tag - instead it has the forward slash.

Reply:
I have a few components like this. Corrected them all.